jmespath-core/src/test/java/io/burt/jmespath/parser/ParserTest.java

```diff
@@ -197,176 +190,151 @@ public void pipesAndChains() {
-      )
+    Expression<Object> expected = Sequence(
+      Property("foo"),
+      Slice(3, 4, 1)
     );
     Expression<Object> actual = compile("foo[3:4]");
     assertThat(actual, is(expected));
   }
 
   @Test
   public void sliceWithoutStopExpression() {
-    Expression<Object> expected = Projection(
-      Current(),
-      Slice(3, null, 1,
-        Property("foo", Current())
-      )
+    Expression<Object> expected = Sequence(
+      Property("foo"),
+      Slice(3, null, 1)
     );
     Expression<Object> actual = compile("foo[3:]");
     assertThat(actual, is(expected));
   }
 
   @Test
   public void sliceWithoutStartExpression() {
-    Expression<Object> expected = Projection(
-      Current(),
-      Slice(null, 4, 1,
-        Property("foo", Current())
-      )
+    Expression<Object> expected = Sequence(
+      Property("foo"),
+      Slice(null, 4, 1)
     );
     Expression<Object> actual = compile("foo[:4]");
     assertThat(actual, is(expected));
   }
 
   @Test
   public void sliceWithStepExpression() {
-    Expression<Object> expected = Projection(
-      Current(),
-      Slice(3, 4, 5,
-        Property("foo", Current())
-      )
+    Expression<Object> expected = Sequence(
+      Property("foo"),
+      Slice(3, 4, 5)
     );
     Expression<Object> actual = compile("foo[3:4:5]");
     assertThat(actual, is(expected));
   }
 
   @Test
   public void sliceWithStepButWithoutStopExpression() {
-    Expression<Object> expected = Projection(
-      Current(),
-      Slice(3, null, 5,
-        Property("foo", Current())
-      )
+    Expression<Object> expected = Sequence(
+      Property("foo"),
+      Slice(3, null, 5)
     );
     Expression<Object> actual = compile("foo[3::5]");
     assertThat(actual, is(expected));
   }
 
   @Test
   public void sliceWithJustColonExpression() {
-    Expression<Object> expected = Projection(
-      Current(),
-      Slice(null, null, 1,
-        Property("foo", Current())
-      )
+    Expression<Object> expected = Sequence(
+      Property("foo"),
+      Slice(null, null, 1)      
     );
     Expression<Object> actual = compile("foo[:]");
     assertThat(actual, is(expected));
   }
 
   @Test
   public void sliceWithJustTwoColonsExpression() {
-    Expression<Object> expected = Projection(
-      Current(),
-      Slice(null, null, 1,
-        Property("foo", Current())
-      )
+    Expression<Object> expected = Sequence(
+      Property("foo"),
+      Slice(null, null, 1)      
     );
     Expression<Object> actual = compile("foo[::]");
     assertThat(actual, is(expected));
   }
 
   @Test
   public void bareSliceExpression() {
-    Expression<Object> expected = Projection(
-      Current(),
-      Slice(0, 1, 2, Current())
-    );
+    Expression<Object> expected = Slice(0, 1, 2);
     Expression<Object> actual = compile("[0:1:2]");
     assertThat(actual, is(expected));
   }
 
   @Test
   public void sliceWithZeroStepSize() {
+    // TODO: SHOULD FAIL?
```

Comment:
Yes it should. I think this test used to be marked with an `@Ignore` to remind about that, but it seems to have gotten lost.